### ISSUE TITLE
Error when using with Windows 10 Pro with Anniversary Update

### ISSUE THREAD

**Issue by jordigg** [puppetlabs/puppetlabs-powershell#140]
Got this error while trying the new version of the module on a Windows10 Pro with the new anniversary update. Tried later on another machine without that update and everything run ok.

Version 1.6.0 works, versions >2.x fail

Any ideas?

```
Debug: 2016-09-02 11:58:15 +0100 C:\Windows\system32\WindowsPowershell\v1.0\powershell.exe -NoProfile -NonInteractive -NoLogo -ExecutionPolicy Bypass -Command - is running as pid: 3532
Debug: Waited 50 milliseconds...
Debug: 2016-09-02 11:58:15 +0100 PIPE> At line:1 char:23

Debug: 2016-09-02 11:58:15 +0100 PIPE> + function New-XmlResult

Debug: 2016-09-02 11:58:15 +0100 PIPE> +                       ~

Debug: 2016-09-02 11:58:15 +0100 PIPE> Missing function body in function declaration.

Debug: 2016-09-02 11:58:15 +0100 PIPE>     + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException

Debug: 2016-09-02 11:58:15 +0100 PIPE>     + FullyQualifiedErrorId : MissingFunctionBody

Debug: 2016-09-02 11:58:15 +0100 PIPE>

Debug: Waited 50 milliseconds...
Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>   param(

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>     [Parameter()]$exitcode,

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>     [Parameter()]$output,

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>     [Parameter()]$errormessage

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>   )

Debug: 2016-09-02 11:58:16 +0100 PIPE> At line:1 char:35

Debug: 2016-09-02 11:58:16 +0100 PIPE> + function Invoke-PowerShellUserCode

Debug: 2016-09-02 11:58:16 +0100 PIPE> +                                   ~

Debug: 2016-09-02 11:58:16 +0100 PIPE> Missing function body in function declaration.

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + FullyQualifiedErrorId : MissingFunctionBody

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE> At line:9 char:11

Debug: 2016-09-02 11:58:16 +0100 PIPE> +     $Code,

Debug: 2016-09-02 11:58:16 +0100 PIPE> +           ~

Debug: 2016-09-02 11:58:16 +0100 PIPE> Missing expression after ','.

Debug: Waited 50 milliseconds...
Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>   # we make our own xml because ConvertTo-Xml makes hard to parse xml ruby side

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>   # and we need to be sure

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>   $xml = [xml]@"

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE> <ReturnResult>

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>   <Property Name='exitcode'>$($exitcode)</Property>

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE> At line:9 char:10

Debug: 2016-09-02 11:58:16 +0100 PIPE> +     $Code,

Debug: 2016-09-02 11:58:16 +0100 PIPE> +          ~

Debug: 2016-09-02 11:58:16 +0100 PIPE> Missing ')' in function parameter list.

Debug: 2016-09-02 11:58:16 +0100 PIPE> At line:1 char:1

Debug: 2016-09-02 11:58:16 +0100 PIPE> + {

Debug: 2016-09-02 11:58:16 +0100 PIPE> + ~

Debug: 2016-09-02 11:58:16 +0100 PIPE> Missing closing '}' in statement block or type definition.

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + FullyQualifiedErrorId : MissingExpressionAfterToken

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: Wait object signaled
Debug: Waited 150 total milliseconds.
Debug: 2016-09-02 11:58:16 +0100 PIPE>   <Property Name='errormessage'>$([System.Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes([string]$error

Debug: 2016-09-02 11:58:16 +0100 PIPE> message)))</Property>

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>   <Property

Debug: 2016-09-02 11:58:16 +0100 PIPE> Name='stdout'>$([System.Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes([string]$output)))</Property>

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE> </ReturnResult>

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE> "@

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>   $xml.OuterXml

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE> IsPublic IsSerial Name                                     BaseType

Debug: 2016-09-02 11:58:16 +0100 PIPE> -------- -------- ----                                     --------

Debug: 2016-09-02 11:58:16 +0100 PIPE> True     True     String                                   System.Object

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE> IsPublic IsSerial Name                                     BaseType

Debug: 2016-09-02 11:58:16 +0100 PIPE> -------- -------- ----                                     --------

Debug: 2016-09-02 11:58:16 +0100 PIPE> True     True     Int32                                    System.ValueType

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE> At line:1 char:16

Debug: 2016-09-02 11:58:16 +0100 PIPE> +     $EventName,

Debug: 2016-09-02 11:58:16 +0100 PIPE> +                ~

Debug: 2016-09-02 11:58:16 +0100 PIPE> Missing expression after ','.

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + FullyQualifiedErrorId : MissingExpressionAfterToken

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE> At line:1 char:3

Debug: 2016-09-02 11:58:16 +0100 PIPE> +   )

Debug: 2016-09-02 11:58:16 +0100 PIPE> +   ~

Debug: 2016-09-02 11:58:16 +0100 PIPE> Unexpected token ')' in expression or statement.

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + FullyQualifiedErrorId : UnexpectedToken

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE> Exception calling "OpenExisting" with "1" argument(s): "Empty name is not legal.

Debug: 2016-09-02 11:58:16 +0100 PIPE> Parameter name: name"

Debug: 2016-09-02 11:58:16 +0100 PIPE> At line:1 char:3

Debug: 2016-09-02 11:58:16 +0100 PIPE> +   $event = [System.Threading.EventWaitHandle]::OpenExisting($EventNam ...

Debug: 2016-09-02 11:58:16 +0100 PIPE> +   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + FullyQualifiedErrorId : ArgumentException

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE> At line:1 char:6

Debug: 2016-09-02 11:58:16 +0100 PIPE> +   try

Debug: 2016-09-02 11:58:16 +0100 PIPE> +      ~

Debug: 2016-09-02 11:58:16 +0100 PIPE> The Try statement is missing its statement block.

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + FullyQualifiedErrorId : MissingTryStatementBlock

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PIPE> At line:163 char:1

Debug: 2016-09-02 11:58:16 +0100 PIPE> + }

Debug: 2016-09-02 11:58:16 +0100 PIPE> + ~

Debug: 2016-09-02 11:58:16 +0100 PIPE> Unexpected token '}' in expression or statement.

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException

Debug: 2016-09-02 11:58:16 +0100 PIPE>     + FullyQualifiedErrorId : UnexpectedToken

Debug: 2016-09-02 11:58:16 +0100 PIPE>

Debug: 2016-09-02 11:58:16 +0100 PowerShell initialization complete for pid: 3532
Debug: Waited 50 milliseconds...
Debug: Waited 50 milliseconds...
Debug: Waited 50 milliseconds...
Debug: Waited 50 milliseconds...
...
```

**Comment by glennsarti** [comment]
Hi @jordigg .  Thankyou for reporting this issue. We have seen this error and we're tracking it at https://tickets.puppetlabs.com/browse/MODULES-3690.  Puppet have two issues logged with the Microsoft Powershell Team and are working to get it fixed.  If you add yourself as a watcher to the MODULES-3690 ticket you'll can get notified of updates.

For the moment I'll close this issue and thanks again for the issue report.
